packages/core/src/awsService/cloudWatchLogs/liveTailLogStreamSubmenu.ts

```diff
@@ -0,0 +1,168 @@
+import { pageableToCollection } from '../../shared/utilities/collectionUtils'
+import { CloudWatchLogs } from 'aws-sdk'
+import { isValidResponse, StepEstimator } from '../../shared/wizards/wizard'
+import { isNonNullable } from '../../shared/utilities/tsUtils'
+
+export enum LogStreamFilterType {
+    MENU = 'menu',
+    PREFIX = 'prefix',
+    SPECIFIC = 'specific',
+    ALL = 'all',
+}
+
+export interface LogStreamFilterResponse {
+    readonly filter?: string
+    readonly type: LogStreamFilterType
+}
+
+export class LogStreamFilterSubmenu extends Prompter<LogStreamFilterResponse> {
+    private logStreamPrefixRegEx = /^[^:*]*$/
+    private currentState: LogStreamFilterType = LogStreamFilterType.MENU
+    private steps?: [current: number, total: number]
+    private region: string
+    private logGroupArn: string
+    public defaultPrompter: QuickPickPrompter<LogStreamFilterType> = this.createMenuPrompter()
+
+    public constructor(logGroupArn: string, region: string) {
+        super()
+        this.region = region
+        this.logGroupArn = logGroupArn
+    }
+
+    public createMenuPrompter() {
+        const helpUri = 'https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_StartLiveTail.html'
+        const prompter = createQuickPick(this.menuOptions, {
+            title: 'Select LogStream filter type',
+            buttons: createCommonButtons(helpUri),
+        })
+        return prompter
+    }
+
+    private get menuOptions(): DataQuickPickItem<LogStreamFilterType>[] {
+        const options: DataQuickPickItem<LogStreamFilterType>[] = []
+        options.push({
+            label: 'All',
+            detail: 'Include log events from all LogStreams in the selected LogGroup',
+            data: LogStreamFilterType.ALL,
+        })
+        options.push({
+            label: 'Specific',
+            detail: 'Include log events from only a specific LogStream',
+            data: LogStreamFilterType.SPECIFIC,
+        })
+        options.push({
+            label: 'Prefix',
+            detail: 'Include log events from LogStreams that begin with a provided prefix',
+            data: LogStreamFilterType.PREFIX,
+        })
+        return options
+    }
+
+    public createLogStreamPrefixBox(): InputBoxPrompter {
+        const helpUri =
+            'https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_StartLiveTail.html#CWL-StartLiveTail-request-logStreamNamePrefixes'
+        return createInputBox({
+            title: 'Enter LogStream prefix',
+            placeholder: 'logStream prefix (case sensitive; empty matches all)',
+            prompt: 'Only log events in the LogStreams that have names that start with the prefix that you specify here are included in the Live Tail session',
+            validateInput: (input) => this.validateLogStreamPrefix(input),
+            buttons: createCommonButtons(helpUri),
+        })
+    }
+
+    public validateLogStreamPrefix(prefix: string) {
```

Comment:
should this have a specific test or do you think its covered by the existing tests you wrote?

Reply:
It should be covered via the existing tests